Content.Shared/Lightning/EntitySystems/SharedMatchstickSystem.cs

```diff
@@ -0,0 +1,85 @@
+using Content.Shared.Interaction;
+using Content.Shared.Item;
+using Content.Shared.Smoking;
+using Content.Shared.Temperature;
+using Robust.Shared.Audio;
+using Robust.Shared.Audio.Systems;
+using Content.Shared.Light.Components;
+using Robust.Shared.Timing;
+
+namespace Content.Shared.Light.EntitySystems;
+
+public abstract class SharedMatchstickSystem : EntitySystem
+{
+    [Dependency] private readonly SharedAppearanceSystem _appearance = default!;
+    [Dependency] private readonly SharedAudioSystem _audio = default!;
+    [Dependency] private readonly SharedItemSystem _item = default!;
+    [Dependency] private readonly SharedPointLightSystem _lights = default!;
+    [Dependency] private readonly IGameTiming _timing = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<MatchstickComponent, InteractUsingEvent>(OnInteractUsing);
+        SubscribeLocalEvent<MatchstickComponent, IsHotEvent>(OnIsHotEvent);
+    }
+
+    private void OnInteractUsing(Entity<MatchstickComponent> ent, ref InteractUsingEvent args)
+    {
+        if (args.Handled || ent.Comp.CurrentState != SmokableState.Unlit)
+            return;
+
+        var isHotEvent = new IsHotEvent();
+        RaiseLocalEvent(args.Used, isHotEvent);
+
+        if (!isHotEvent.IsHot)
+            return;
+
+        Ignite(ent, args.User);
+        args.Handled = true;
+    }
+
+    private void OnIsHotEvent(EntityUid uid, MatchstickComponent component, IsHotEvent args)
+    {
+        args.IsHot = component.CurrentState == SmokableState.Lit;
```

Comment:
```suggestion
        args.IsHot |= component.CurrentState == SmokableState.Lit;
```
may overwrite another event subscription that already set it to true